state-chain/runtime/src/chainflip/solana_elections.rs

```diff
@@ -51,6 +52,338 @@ pub type SolanaElectoralSystem = Composite<
+				old_nonce_identifiers,
+				&(),
+			)?;
+			Ok(())
+		}
+	}
+	#[derive(Encode, Decode, TypeInfo, Clone)]
+	struct ElectionBitmapComponents {
+		epoch: EpochIndex,
+		#[allow(clippy::type_complexity)]
+		bitmaps: Vec<(
+			<<SolanaElectoralSystem as ElectoralSystem>::Vote as VoteStorage>::BitmapComponent,
+			BitVec<u8, bitvec::order::Lsb0>,
+		)>, //sp_core::H256, BitVec<u8, bitvec::order::Lsb0>)>,
+	}
+	#[derive(PartialEq, Eq, Clone, Debug, Encode, Decode, TypeInfo, Default)]
+	pub struct ReferenceDetails {
+		pub count: u32,
+		pub created: u32,
+		pub expires: u32,
+	}
+	pub struct Migration;
+	impl OnRuntimeUpgrade for Migration {
+		#[cfg(feature = "try-runtime")]
+		fn pre_upgrade() -> Result<Vec<u8>, DispatchError> {
+			let election_identifiers = frame_support::migration::storage_key_iter::<
+				ElectionIdentifierOf<old::SolanaElectoralSystem>,
+				<old::SolanaElectoralSystem as ElectoralSystem>::ElectionProperties,
+				Twox64Concat
+			>(b"SolanaElections", b"ElectionProperties")
+				.filter(|(_, value)| {
+					matches!(value, pallet_cf_elections::electoral_systems::composite::tuple_6_impls::CompositeElectionProperties::D(_))
+				})
+				.map(|(key, value)| {
+					log::info!("Old {:?}: {:?}",key, value);
+					key
+				})
+				.collect::<Vec<ElectionIdentifierOf<old::SolanaElectoralSystem>>>();
+			log::info!("Number of elections: {:?}", election_identifiers.len());
+			Ok((election_identifiers.len() as u32).encode())
+		}
+		#[cfg(feature = "try-runtime")]
+		fn post_upgrade(state: Vec<u8>) -> Result<(), DispatchError> {
+			let previous_number_election = u32::decode(&mut &state[..]).unwrap();
+			log::info!("Post upgrade number of election old state: {:?}", previous_number_election);
+			log::info!(
+				"Post upgrade number of unavailable nonces: {:?}",
+				pallet_cf_environment::SolanaUnavailableNonceAccounts::<Runtime>::iter_keys()
+					.collect::<Vec<_>>()
+					.len() as u32
+			);
+
+			assert!(
+				previous_number_election ==
+					pallet_cf_environment::SolanaUnavailableNonceAccounts::<Runtime>::iter_keys()
+						.collect::<Vec<_>>()
+						.len() as u32
+			);
+			let election_identifiers = frame_support::migration::storage_key_iter::<
+				ElectionIdentifierOf<super::SolanaElectoralSystem>,
+				<super::SolanaElectoralSystem as ElectoralSystem>::ElectionProperties,
+				Twox64Concat
+			>(b"SolanaElections", b"ElectionProperties")
+				.filter(|(_, value)| {
+					matches!(value, pallet_cf_elections::electoral_systems::composite::tuple_6_impls::CompositeElectionProperties::D(_))
+				})
+				.map(|(key, _)| {
+					key
+				})
+				.collect::<Vec<ElectionIdentifierOf<super::SolanaElectoralSystem>>>();
+			log::info!("Post upgrade number of elections: {:?}", election_identifiers.len() as u32);
+			assert!(previous_number_election == election_identifiers.len() as u32);
+
+			Ok(())
+		}
+		fn on_runtime_upgrade() -> frame_support::weights::Weight {
```

Comment:
can we make sure we find some selection of blocks on mainnet that have elections of this type and run try-runtime against them

Reply:
I tested it against some specific blocks with some unavailable nonces as well as during bouncer (to be sure to have a lot of unavailable nonces) and it passed everything.
I am unable to test it against mainnet though, the command always times out.. Here is a block with many unavailable nonces:
`0x254f6d74ae97f08681b448169b43cdd1363c576f0a713e6f8e2ef06371c4c0ca` -> 9 nonces unavailable